src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
+            throw new IndexOutOfBoundsException("given index does not exist");
+        } else {
+            int currentIndex = 0;
+            Node currentNode = head;
+            while (currentIndex != index) {
+                currentNode = currentNode.next;
+                currentIndex++;
+            }
+            T previousNode = (T) currentNode.value;
+            currentNode.value = value;
+            return previousNode;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index > size - 1 || index < 0) {
+            throw new IndexOutOfBoundsException("given index does not exist");
+        } else if (size == 1) {
+            Node removedNode = head;
+            head = null;
+            size--;
+            return (T) removedNode.value;
+        } else if (index == 0) {
+            Node removedNode = head;
+            head.next.prev = null;
+            head = head.next;
+            size--;
+            if (removedNode == null) {
+                return null;
+            }
+            return (T) removedNode.value;
+        } else if (index == size - 1) {
+            Node removedNode = tail;
+            tail.prev.next = null;
+            tail = tail.prev;
+            size--;
+            return (T) removedNode.value;
+        } else {
+            int currentIndex = 0;
+            Node currentNode = head;
+            while (currentIndex != index) {
+                currentNode = currentNode.next;
+                currentIndex++;
+            }
+            Node removedNode = currentNode;
+            currentNode.prev.next = currentNode.next;
+            currentNode.next.prev = currentNode.prev;
+            size--;
+            return (T) removedNode.value;
+        }
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node currentNode = head;
+        int index = 0;
+        int indexOfRemovingObject = -1;
+        do {
+            if ((object != null && currentNode.value != null && object.equals(currentNode.value))
+                || object == null && currentNode.value == null) {
+                indexOfRemovingObject = index;
+                break;
+            }
+            index++;
+            currentNode = currentNode.next;
+        } while (index != size - 1);
+
+        if (indexOfRemovingObject != -1) {
+        remove(indexOfRemovingObject);
+        }
+        return indexOfRemovingObject != -1;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
     }
```

Comment:
Make Node<T> an inner class. The purpose of inner classes is to group classes that belong together, which makes your code more readable and maintainable. Inner class should be at the end of the class.